### PR TITLE
[MI-2634]:Fixed github issue #558 on duplicate notifications

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
https://github.com/mattermost/mattermost-plugin-github/issues/558

### CODE REVIEW

File: server/plugin/webhook.go [hunk 1]
```diff
@@ -1030,7 +1044,29 @@ func (p *Plugin) handleCommentAssigneeNotification(event *github.IssueCommentEve
...
+		usernameMentioned := false
+		for _, username := range mentionedUsernames {
+			if username == *assignee.Login {
+				usernameMentioned = true
+				break
+			}
+		}
+
+		if usernameMentioned {
+			switch splitURL[len(splitURL)-2] {
```
Comment (ayusht2810) [inline server/plugin/webhook.go]: have we already checked for its length?

File: server/plugin/webhook.go [hunk 2]
```diff
@@ -1030,7 +1044,29 @@ func (p *Plugin) handleCommentAssigneeNotification(event *github.IssueCommentEve
...
+			case "pull":
+				templateName = "commentAssigneeSelfMentionPullRequestNotification"
+			case "issues":
+				templateName = "commentAssigneeSelfMentionIssueNotification"
+			default:
+				p.API.LogWarn("Unhandled issue type", "type", splitURL[len(splitURL)-2])
```
Comment (ayusht2810) [inline server/plugin/webhook.go]:
```suggestion
				p.API.LogWarn("Unhandled issue type", "Type", splitURL[len(splitURL)-2])
```

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin/webhook.go (hunk 1 above)
Comment (Kshitij-Katiyar) [reply server/plugin/webhook.go]: we have to check for this special case whether this is a pull request or an issue.

File: server/plugin/webhook.go [hunk 3]
```diff
@@ -1033,14 +1033,15 @@ func (p *Plugin) handleCommentAssigneeNotification(event *github.IssueCommentEve
...
 	var templateName string
-	switch splitURL[len(splitURL)-2] {
+	switch eventType {
 	case "pull":
 		templateName = "commentAssigneePullRequestNotification"
 	case "issues":
 		templateName = "commentAssigneeIssueNotification"
 	default:
-		p.API.LogWarn("Unhandled issue type", "type", splitURL[len(splitURL)-2])
+		p.API.LogWarn("Unhandled issue type", "type", eventType)
```
Comment (ayusht2810) [inline server/plugin/webhook.go]:
```suggestion
		p.API.LogWarn("Unhandled issue type", "Type", eventType)
```

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin/webhook.go (hunk 1 above)
Comment (ayusht2810) [reply server/plugin/webhook.go]: I was talking about the length if it's smaller than 2. But it's already being checked.

File: server/plugin/webhook.go [hunk 4]
```diff
@@ -1019,18 +1033,41 @@ func (p *Plugin) handleCommentAssigneeNotification(event *github.IssueCommentEve
 	if len(splitURL) < 2 {
 		return
 	}
```
Comment (manojmalik20) [inline server/plugin/webhook.go:1035]:
```suggestion
	}
	
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**